ccvm_simulators/solvers/dl_solver.py

```diff
@@ -103,6 +103,7 @@ def _method_selector(self, problem_category):
         """
         if problem_category.lower() == "boxqp":
             self.calculate_grads = self._calculate_grads_boxqp
+            self.calculate_grads_adam = self._calculate_grads_boxqp_adam
```

Comment:
I don't think we need both of these methods, but I'll wait for your reply to my comment on `__call__` because that could change things

Reply:
Pls see the [comment](https://github.com/1QB-Information-Technologies/ccvm/pull/99#discussion_r1323409756)

Reply:
Thank you. This makes sense to me.